Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -768,10 +768,7 @@ namespace Babylon
         const auto& vertexArray = *(info[0].As<Napi::External<VertexArray>>().Data());
 
         // a vertex array might not have an index buffer associated with
-        if (vertexArray.indexBuffer.data)
-        {
-            vertexArray.indexBuffer.data->SetBgfxIndexBuffer();
-        }
+        m_currentBoundIndexBuffer = vertexArray.indexBuffer.data;
```

Comment:
Given the way things are going, I wonder if it would be cleaner to store the vertexArray instead of just the index buffer. Then do the bgfx work when draw is called.

Reply:
Not sure if a vertex/indexbuffer is bound once with multiple drawcalls depending on it. But maybe in the future. I see a trend to store a lot more states in nativeengine class (clear/rendering states/index buffer).

Reply:
We should probably take care of this sooner than later.